internal/transformations/hex_decode.go

```diff
@@ -0,0 +1,26 @@
+package transformations
+
+import (
+	"encoding/hex"
+
+	"github.com/corazawaf/coraza/v3/internal/strings"
+)
+
+func hexDecode(data string) (string, bool, error) {
+	src := []byte(data)
+
+	// According to RFC-4648 section 8, the valid length of src MUST be even.
+	// Here https://datatracker.ietf.org/doc/html/rfc4648#section-8
+	// There was a decision to cut "redundant" bytes for the "best effort approach" and proceed decoding.
```

Comment:
What does this mean? Can you provide pointers?

"hex" enconding, in the RFC you are mentioning, is just Base16 where each byte is represented as two four bits.

So, AFAIU, we are only removing the _last_ input to make an effort to fix a broken input, right? Could the problem be the first instead?

Reply:
This PR is mostly based on assumptions cause I didn't get any reply here https://github.com/corazawaf/coraza/issues/1253#issuecomment-2556942463

The main assumptoin that coraza want to go "best effort approach". This assumption is based on the tests. The tests made so that we put a specific input and get a specific output. Relying on tests we need to remove the last symbol. 
This is not a part of RFC (it can look like I say it is in the comment, I agree. I should change the comment).

If you ask me I'd go "fail fast" approach where we are not trying to guess which symbol is wrong. But I'm not fully in the context so probably ""fail fast" is not applicable for coraza.